test/workload-deployment.sh

```diff
@@ -0,0 +1,140 @@
+echo "-------------------------------------------------------------------------"
+kubectl --context wds1 apply -f - <<EOF
+apiVersion: edge.kubestellar.io/v1alpha1
+kind: Placement
+metadata:
+  name: nginx-placement
+spec:
+  clusterSelectors:
+  - matchLabels: {"location-group":"edge"}
+  downsync:
+  - labelSelectors:
+    - matchLabels: {"app.kubernetes.io/name":"nginx"}
+EOF
+
+echo "Deploy the app"
+echo "-------------------------------------------------------------------------"
+kubectl --context wds1 apply -f - <<EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  labels:
+    app.kubernetes.io/name: nginx
+  name: nginx
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  namespace: nginx
+  labels:
+    app.kubernetes.io/name: nginx
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: public.ecr.aws/nginx/nginx:latest 
+        ports:
+        - containerPort: 80
+EOF
+
+echo "Verify that manifestworks wrapping the objects have been created in the mailbox namespaces"
+echo "-------------------------------------------------------------------------"
+kubectl --context imbs1 get manifestworks -n cluster1
+kubectl --context imbs1 get manifestworks -n cluster2
+
+echo "Verify that the deployment has been created in both clusters"
+echo "-------------------------------------------------------------------------"
+function wait_for_namespace() {
+  # kubectl wait can't be used on resources that haven't been created, so we need to spin
+  cluster=$1
+  echo "Waiting for deployment on $cluster"
+  waitCounter=0
+  while (($(kubectl --context $cluster wait namespace nginx --for jsonpath='{.status.phase}=Active' --timeout=180s 2>/dev/null | grep -c "condition met") < 1)); do
+    if (($waitCounter > 180)); then
+      echo "Failed to observe namespace on ${cluster}."
+      exit 1 
+    fi
+    ((waitCounter += 1))
+    sleep 1
+  done
+}
+wait_for_namespace cluster1
+kubectl --context cluster1 wait deployment nginx-deployment -n nginx --for=condition=available --timeout=180s
+echo "Waiting for deployment on cluster2"
+wait_for_namespace cluster2
+kubectl --context cluster2 wait deployment nginx-deployment -n nginx --for=condition=available --timeout=180s
+echo "SUCCESS: confirmed deployments on both cluster1 and cluster2."
+
+echo "Cleanup"
+echo "-------------------------------------------------------------------------"
+kind delete cluster --name cluster1
+kind delete cluster --name cluster2
+kind delete cluster --name kubeflex
+kubectl config delete-context cluster1 
+kubectl config delete-context cluster2 
```

Comment:
Should we also clean up the contexts for imbs1 and wds1?

Reply:
Removing wds1 is simple, but imbs1 is the current context.  What should we switch the context to if there are no other contexts?  Remove the kube config file all together?  If we don't delete imbs1 at all, then the next invocation of the test will overwrite it, which perhaps is also the wrong behaviour?

Reply:
Maybe let the test somehow remember the original/initial 'current context' at the beginning of the test (maybe something like `original_ctx="$(kubectl config current-context)"`), and restore it (if not empty) at the end? I think leaving an invalid context in user's kubeconfig is strange. But it's just my opinion and totally up to you.

Reply:
I have trouble even before that:

```
...
kubectl config rename-context kind-${CLUSTER} ${CLUSTER}
...
error: cannot rename the context "kind-cluster1", the context "cluster1" already exists in /Users/mspreitz/.kube/config
...
```

Reply:
That happened Mike because you already have a cluster1 in your kubeconfig.  Likely a previous run failed and the cleanup was not ran?  How would you like such a scenario to be handled in the test - fail as it currently does or force a cleanup even though cluster1 might be an unrelated cluster that the user created?

Reply:
I think each test should begin with deleting stuff that would cause problems.

Reply:
So delete the kind clusters named kubeflex, cluster1, cluster2, and remove all the relevant contexts? It's a bit destructive although you could argue that kind clusters don't usually host real workloads. Alternatively, how about running the cleanup script prior to the test explicitly rather than implicitly remove clusters?

Reply:
Deleting stuff at the _end_ of the test is positively bad, it prevents the user from building on the state established by the test or even examining it to learn things. That is why I suggested doing the deletions at the _start_.

The test _is_ destructive of the things it involves, so I think it is fine to delete them at the _start_ of the test.